lambdas/services/process_mns_message_service.py

```diff
@@ -0,0 +1,140 @@
+            )
+            logger.info(f"{e}")
+            return
+
+    def handle_gp_change_notification(self, message: MNSSQSMessage):
+        patient_document_references = self.get_patient_documents(
+            message.subject.nhs_number
+        )
+
+        if len(patient_document_references) < 1:
+            logger.info("Patient is not held in the National Document Repository.")
+            logger.info("Moving on to the next message.")
+            return
+
+        updated_ods_code = self.get_updated_gp_ods(message.subject.nhs_number)
+
+        for reference in patient_document_references:
+            if reference["CurrentGpOds"] is not updated_ods_code:
+                self.dynamo_service.update_item(
+                    table_name=self.table,
+                    key=reference["ID"],
+                    updated_fields={"CurrentGpOds": updated_ods_code},
+                )
+
+        logger.info("Update complete for change of GP")
+
+    def handle_death_notification(self, message: MNSSQSMessage):
+        if self.is_informal_death_notification(message):
+            logger.info(
+                "Patient is deceased - INFORMAL, moving on to the next message."
+            )
+            return
+
+        try:
+            patient_documents = self.get_patient_documents(message.subject.nhs_number)
+
+            if len(patient_documents) < 1:
+                logger.info("Patient is not held in the National Document Repository.")
+                logger.info("Moving on to the next message.")
+                return
+
+            if (
+                message.data["deathNotificationStatus"]
+                == DeathNotificationStatus.FORMAL.value
+            ):
+                self.update_patient_details(
+                    patient_documents, PatientOdsInactiveStatus.DECEASED.value
+                )
+                logger.info("Update complete, patient marked DECE.")
+                return
+
+            if (
+                message.data["deathNotificationStatus"]
+                == DeathNotificationStatus.REMOVED.value
+            ):
+                update_ods_code = self.get_updated_gp_ods(message.subject.nhs_number)
+                self.update_patient_details(patient_documents, update_ods_code)
+                logger.info("Update complete for death notification change.")
+        except ClientError as e:
+            logger.error(f"{e}")
+
+    def is_informal_death_notification(self, message: MNSSQSMessage) -> bool:
+        return (
+            message.data["deathNotificationStatus"]
+            == DeathNotificationStatus.INFORMAL.value
+        )
+
+    def get_patient_documents(self, nhs_number: str):
+        logger.info("Getting patient document references...")
+        response = self.dynamo_service.query_table_by_index(
+            table_name=self.table,
+            index_name="NhsNumberIndex",
+            search_key="NhsNumber",
+            search_condition=nhs_number,
+        )
```

Comment:
Maybe also filter the record wasn't deleted? Do we want to update if it was?

Reply:
@steph-torres-nhs was this addressed? It still looks like it's not filtering from what I can see

Reply:
Yeah this was addressed after a discussion with Ifey, keeping as is for time being